### DISCUSSION TITLE
vitest test runner

### DISCUSSION THREAD

**Discussion by capaj** [romeerez/orchid-orm#37] [Q&A]
Hey what do you think about vitest?
Would you be open to accept a PR which would refactor tests to run in vitest?

**Comment by romeerez** [comment]
Hey, out of curiosity, I tried Vitest already here in `pqb` module, and decided to stay with Jest because of this: https://github.com/vitest-dev/vitest/issues/579

Vitest is almost identical to Jest but performs slower.

**Reply by capaj** [reply]
Slower? My experience with Vitest has been the opposite. I run a huge testsuite on top of postgres with it every day. I switched  to vitest last year from jest due to vitest being much faster.

**Reply by capaj** [reply]
So would you be open to merging a PR if it ran faster than jest?

**Reply by romeerez** [reply]
> So would you be open to merging a PR if it ran faster than jest?

Yes, absolutely, but I have warned you :)

Vitest is faster if you don't setup Jest to use `swc`, but when you do, Vitest is 2-3 times slower by default according to my own experience just around a month ago in `pqb`, and 1.5 times slower when threads are turned off.

**Reply by romeerez** [reply]
There are 1.3k tests in `pqb` and Jest shows < 5s running them all.

**Reply by capaj** [reply]
whoa that is fast.

I tried running the specs, but I think I need to initialize the DB somehow?
I am getting a lot of 

```
error: relation "user" does not exist
```

**Reply by romeerez** [reply]
Yeah, I still didn't write a Contribution guide

So, copy .env.example to .env in the root, set PG_URL, (MYSQL_URL can be ignored),
go to packages/rake-db, run `pnpm db create` and then `pnpm db migrate`
and after that it should run the tests
